### DISCUSSION TITLE
testconsole - Manual Generation without pipeline

### DISCUSSION THREAD

**Discussion by edwards2018** [rfennell/AzurePipelines#1198] [Q&A] [answered]
Good morning, I'm not sure if I am asking the impossible here, but is there currently the facility to use the test console with a different input than buildID.....I have some projects that are not yet pipelined but would like to perform a similar extract using the same template- but passing in a payload that is based on a last commit ID for example.

This would not be a long term solution, but would enable me to use this tool where pipeline creation is still in flight

**Comment by rfennell** [comment, marked as answer]
The problem is all the logic in this task (and hence tester) is based around a buildID or releaseID. All the WI/CS associations are found by their links to a pipeline run using Microsoft provided REST API calls. 

So sorry what you are asking for is not possible without a complete rewrite.

However, the workaround I would suggest is to just create a temporary pipeline. It does not have to do anything at all, other than be associated with the repo containing the source. You can then report either by a task in the pipeline or with the tester

Hope that helps

**Reply by edwards2018** [reply]
Thank you, I thought that might be the case, and I was considering the workaround approach to bridge the gap in the interim. Thank you for prompt feedback, really helpful
